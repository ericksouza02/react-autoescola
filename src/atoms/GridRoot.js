import styled, {css} from "styled-components";
import PropTypes from 'prop-types';

import {breakAt, BreakPointSizes} from '../styles/BreakPointSize'

const breakColumnAt = (size) => props => props[size] && css`
${breakAt(BreakPointSizes[size])}{
grid-template-columns: repeat(${props => props[size]}, 1fr);
}
` 

const GridRoot = styled.div`
    display: grid;
    grid-column-gap: 16px;
    grid-row-gap: 16px;
    grid-template-columns: 1fr;

    ${breakColumnAt("sm")}
    ${breakColumnAt("md")}
    ${breakColumnAt("lg")}
    ${breakColumnAt("xl")}
  
`

GridRoot.defaultProps = {
    sm: undefined,
    md: undefined,
    lg: undefined,
    xl: undefined,
}

GridRoot.propTypes = {
    sm: PropTypes.number,
    md: PropTypes.number,
    lg: PropTypes.number,
    xl: PropTypes.number,
}

export default GridRoot;