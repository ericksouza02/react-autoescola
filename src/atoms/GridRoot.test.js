import React from "react";
import {render} from '@testing-library/react';

import GridRoot from "./GridRoot"

test('match snapshot with no params', () => {
    const {asFragment} = render(<GridRoot />)
    

    expect(asFragment()).toMatchSnapshot()
})

test('match snapshot with params', () => {
    const {asFragment} = render(<GridRoot sm={2} md={3} lg={4} xl={5}/>)
    

    expect(asFragment()).toMatchSnapshot()
})