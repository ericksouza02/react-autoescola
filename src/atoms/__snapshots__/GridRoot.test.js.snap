// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot with no params 1`] = `
<DocumentFragment>
  <div
    class="sc-beyTiQ lbZvPe"
  />
</DocumentFragment>
`;

exports[`match snapshot with params 1`] = `
<DocumentFragment>
  <div
    class="sc-beyTiQ kJUkUC"
    lg="4"
    md="3"
    sm="2"
    xl="5"
  />
</DocumentFragment>
`;
